From adc3b0502218d7469b7020b85cc54241a22be274 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 25 Nov 2018 18:23:53 +0200
Subject: [PATCH 0217/1239] arm64: mvebu: espressobin: add support for the
 variant with eMMC

Add support for additional EspressoBIN board with installed
eMMC device (U11).
Starting from this patch the DEVICE_TREE= must be added to
"make" in order to distinguish between platforms wth and without
eMMC on board.
Regualr (no eMMC) EspressoBIN builds should use DTS file named
armada-3720-espressobin and build for boards with eMMC installed
the DTS named armada-3720-espressobin-emmc.
The default device tree string is now removed from
mvebu_espressobin-88f3720_defconfig config file.
Update build documentation accordingly.

Change-Id: Idc6b79852316fdfc099c3e79e5ae9a27e4a80a42
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
---
 arch/arm/dts/Makefile                         |  1 +
 arch/arm/dts/armada-3720-espressobin-emmc.dts | 19 +++++++++++++++++++
 arch/arm/dts/armada-3720-espressobin.dts      |  2 ++
 configs/mvebu_espressobin-88f3720_defconfig   |  1 -
 4 files changed, 22 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/dts/armada-3720-espressobin-emmc.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 1a3bb07758..a2be46e820 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -181,6 +181,7 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-3720-ddr4-db-v1-A.dtb		\
 	armada-3720-ddr4-db-v1-B.dtb		\
 	armada-3720-espressobin.dtb		\
+	armada-3720-espressobin-emmc.dtb	\
 	armada-3720-turris-mox.dtb		\
 	armada-3720-uDPU.dtb			\
 	armada-375-db.dtb			\
diff --git a/arch/arm/dts/armada-3720-espressobin-emmc.dts b/arch/arm/dts/armada-3720-espressobin-emmc.dts
new file mode 100644
index 0000000000..0dd59af9c0
--- /dev/null
+++ b/arch/arm/dts/armada-3720-espressobin-emmc.dts
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2018 Marvell International Ltd.
+ */
+
+#include "armada-3720-espressobin.dts"
+
+/ {
+	model = "Marvell Armada 3720 Community Board ESPRESSOBin (eMMC)";
+	compatible = "marvell,armada-3720-espressobin",
+		     "marvell,armada-3720-espressobin-emmc",
+		     "marvell,armada3720", "marvell,armada3710";
+
+};
+
+/* U11 */
+&sdhci1 {
+	status = "okay";
+};
diff --git a/arch/arm/dts/armada-3720-espressobin.dts b/arch/arm/dts/armada-3720-espressobin.dts
index 1709426a66..da6576f73c 100644
--- a/arch/arm/dts/armada-3720-espressobin.dts
+++ b/arch/arm/dts/armada-3720-espressobin.dts
@@ -137,6 +137,7 @@
 	status = "okay";
 };
 
+/* J1 */
 &sdhci0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdio_pins>;
@@ -145,6 +146,7 @@
 	status = "okay";
 };
 
+/* U11 */
 &sdhci1 {
 	non-removable;
 	bus-width = <8>;
diff --git a/configs/mvebu_espressobin-88f3720_defconfig b/configs/mvebu_espressobin-88f3720_defconfig
index 67031593e8..6aefa391bb 100644
--- a/configs/mvebu_espressobin-88f3720_defconfig
+++ b/configs/mvebu_espressobin-88f3720_defconfig
@@ -34,7 +34,6 @@ CONFIG_CMD_TIME=y
 CONFIG_CMD_MVEBU_BUBT=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_MAC_PARTITION=y
-CONFIG_DEFAULT_DEVICE_TREE="armada-3720-espressobin"
 CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_AHCI_MVEBU=y
 CONFIG_CLK=y
-- 
2.29.0

